File: webrtc/src/main/java/com/bandwidth/webrtc/signaling/websockets/WebSocketProvider.java
package com.bandwidth.webrtc.signaling.websockets;

import com.bandwidth.webrtc.signaling.websockets.listeners.OnCloseListener;
import com.bandwidth.webrtc.signaling.websockets.listeners.OnErrorListener;
import com.bandwidth.webrtc.signaling.websockets.listeners.OnMessageListener;
import com.bandwidth.webrtc.signaling.websockets.listeners.OnOpenListener;

public interface WebSocketProvider {
    void setOnOpenListener(OnOpenListener onOpenListener);
    void setOnCloseListener(OnCloseListener onCloseListener);
    void setOnMessageListener(OnMessageListener onMessageListener);
    void setOnErrorListener(OnErrorListener onErrorListener);

    void open() throws WebSocketException;
    void close();
    void sendMessage(String message);
}

File: webrtc/src/main/java/com/bandwidth/webrtc/signaling/websockets/WebSocketProvider.kt
package com.bandwidth.webrtc.signaling.websockets

import com.bandwidth.webrtc.signaling.websockets.listeners.OnOpenListener
import com.bandwidth.webrtc.signaling.websockets.listeners.OnMessageListener
import com.bandwidth.webrtc.signaling.ConnectionException
import com.bandwidth.webrtc.signaling.websockets.listeners.OnCloseListener
import com.bandwidth.webrtc.signaling.websockets.listeners.OnErrorListener

interface WebSocketProvider {
    fun setOnOpenListener(onOpenListener: OnOpenListener)
    fun setOnCloseListener(onCloseListener: OnCloseListener)
    fun setOnMessageListener(onMessageListener: OnMessageListener)
    fun setOnErrorListener(onErrorListener: OnErrorListener)

    @Throws(ConnectionException::class)
    fun open(uri: String?)
    fun close()
    fun sendMessage(message: String?)
}